shared/src/commonMain/kotlin/org/hyperskill/app/topics_repetitions/presentation/TopicsRepetitionsReducer.kt

```diff
@@ -50,7 +56,10 @@ class TopicsRepetitionsReducer : StateReducer<State, Message, Action> {
             is Message.NextTopicsRepetitionsLoaded.Success ->
                 if (state is State.Content) {
                     state.copy(
-                        topicsRepetitions = state.topicsRepetitions + message.nextTopicsRepetitions,
+                        topicsRepetitions = state.topicsRepetitions +
+                            message.nextTopicsRepetitions.filter { nextTopicRepetitions ->
+                                state.topicsRepetitions.none { it.topicId == nextTopicRepetitions.topicId }
+                            },
```

Comment:
```suggestion
                        topicsRepetitions = state.topicsRepetitions.union(message.nextTopicsRepetitions).toList(),
```
You can replace `plus` and `filter` operation with just one operator [`union`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/union.html) witch _returns a set containing all distinct elements from both collections_